source/Calamari.AzureAppService/Behaviors/Legacy/LegacyAzureAppServiceBehaviour.cs

```diff
@@ -0,0 +1,216 @@
+            Log.Verbose($@"Publishing {uploadZipPath} to {zipUploadUrl}");
+
+            // The HttpClient default timeout is 100 seconds: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-5.0#remarks
+            // This timeouts with even relatively small packages: https://octopus.zendesk.com/agent/tickets/69928
+            // We'll set this to an hour for now, but we should probably implement some more advanced retry logic, similar to https://github.com/OctopusDeploy/Sashimi.AzureWebApp/blob/bbea36152b2fb531c2893efedf0330a06ae0cef0/source/Calamari/AzureWebAppBehaviour.cs#L70
+            client.Timeout = TimeSpan.FromHours(1);
+
+            //we add some retry just in case the web app's Kudu/SCM is not running just yet
+            var response = await RetryPolicies.TransientHttpErrorsPolicy.ExecuteAsync(async () =>
+                                                                                      {
+                                                                                          //we have to create a new request message each time
+                                                                                          var request = new HttpRequestMessage(HttpMethod.Post, zipUploadUrl)
+                                                                                          {
+                                                                                              Headers =
+                                                                                              {
+                                                                                                  Authorization = new AuthenticationHeaderValue("Basic", publishingProfile.GetBasicAuthCredentials())
+                                                                                              },
+                                                                                              Content = new StreamContent(new FileStream(uploadZipPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+                                                                                              {
+                                                                                                  Headers = { ContentType = new MediaTypeHeaderValue("application/octet-stream") }
+                                                                                              }
+                                                                                          };
+
+                                                                                          var r = await client.SendAsync(request);
+                                                                                          r.EnsureSuccessStatusCode();
+                                                                                          return r;
+                                                                                      });
```

Comment:
I understand this is not the main change of this PR, but do we want to keep the legacy class exactly the same as before? I have no strong opinions here.

Reply:
I did originally, but then I started getting flaky tests, so I introduced this into the legacy code to resolve that